servicebus/data-plane/azure-servicebus/src/main/java/com/microsoft/azure/servicebus/TransactionContext.java

```diff
@@ -36,7 +38,7 @@ public TransactionContext(ByteBuffer txnId, MessagingFactory messagingFactory) {
 
     @Override
     public String toString() {
-        return new String(txnId.array(), txnId.position(), txnId.limit());
+        return new String(txnId.array(), txnId.position(), txnId.limit(), UTF_8);
```

Comment:
[SpotBugs P1] Reliance on default encoding: Found a call to a method which will perform a byte to String (or String to byte) conversion, and will assume that the default platform encoding is suitable. This will cause the application behavior to vary between platforms. Use an alternative API and specify a charset name or Charset object explicitly.